components/malloc_size_of/lib.rs

```diff
@@ -782,3 +791,91 @@ impl MallocSizeOf for xml5ever::QualName {
         self.local.size_of(ops)
     }
 }
+
+impl MallocSizeOf for hyper::header::Headers {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let mut t = 0;
+        for x in self.iter() {
+           t += x.size_of(ops)
+        }
+        t
+    }
+}
+
+impl<'a> MallocSizeOf for hyper::header::HeaderView<'a> {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // Close enough?
+        self.name().size_of(ops) + self.value_string().size_of(ops)
```

Comment:
This won't do what we want, unfortunately. Instead, let's do `self.0.size_of(ops)` and implement MallocSizeOf for `Unicase<T>` where `T: MallocSizeOf`, and add `self.1.raw_mut().size_of(ops)`.

Reply:
The compiler is complaining about private fields for HeaderView - I remember trying this as an attempt to skirt around the issue. 

I attempted to implement MallocSizeOf for UniCase<T> (from the unicase crate, I assumed!) using `self.1.raw_mut().size_of(ops)` I got an out-of-bounds index, and looking through the docs it doesn't immediately appear that UniCase has any fields.

Any advice would be helpful!!

Reply:
Ok, I understand better now. In that case, rather than implementing MallocSizeOf for HeaderView, we can change the implementation for Headers to this:
```rust
for x in self.iter() {
    let name = x.name();
    let raw = self.get_raw(name);
    t += raw.size_of(ops);
}
```
This will undercount, but it's better than nothing!